sources/Renderer/RenderContext.cpp

```diff
@@ -180,9 +152,15 @@ bool RenderContext::SetVideoModePrimary(const VideoModeDescriptor& videoModeDesc
 
     auto& surface = GetSurface();
 
+    // XXX TODO We should not assume that a surface can enter fullscreen or even
```

Comment:
The manual store and restore of the surface location is indeed not a very elegant solution. But again, please don't just disable those code paths and please try to keep the PR small.

Reply:
I believe this will actively interfere with the way the Mac enters and exists borderless fullscreen mode. I think the facility should be present but called on by the platform code instead of the core code.